### PR TITLE
Pull Request for Issue2173: adding basic Ocean Optics detector functionality

### PR DESCRIPTION
**Pull request by helfrij**
- Created class PGMOceanOpticsXRFDetector with basic acquisition functionality.
- Added view for the detector to PGMAppController.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you review?

File: source/application/PGM/PGMAppController.cpp [hunk 1]
```diff
@@ -103,7 +104,9 @@ void PGMAppController::createGeneralPanes()
 
 void PGMAppController::createDetectorPanes()
 {
-
+	AMXRFDetailedDetectorView *oceanOpticsDetectorView = new AMXRFDetailedDetectorView(PGMBeamline::pgm()->oceanOpticsDetector());
+	oceanOpticsDetectorView->buildDetectorView();
+	mw_->addPane(oceanOpticsDetectorView, detectorPaneCategoryName_, "Ocean Optics", generalPaneIcon_);
```
Comment (XuetaoLiu-CLS) [inline source/application/PGM/PGMAppController.cpp]: the icon should be `detectorPaneIcon_`

File: source/beamline/PGM/PGMBeamline.cpp [hunk 2]
```diff
@@ -147,6 +169,18 @@ void PGMBeamline::setupExposedDetectors()
 	addExposedDetector(i0EndstationBladeCurrentDetector_);
 	addExposedDetector(i0BeamlineBladeCurrentDetector_);
 	addExposedDetector(photodiodeBladeCurrentDetector_);
+
+	addExposedDetector(oceanOpticsDetector_);
+}
+
+bool PGMBeamline::getConnectedState() const
```
Comment (XuetaoLiu-CLS) [inline source/beamline/PGM/PGMBeamline.cpp]: in another PR of mine, this function is called `isConnected()` and I think that is a common name we used for the other components.

Comment (XuetaoLiu-CLS) [inline source/beamline/PGM/PGMBeamline.cpp]: never mind... there is a isConnected() function, which returns the `connected_` flag. 

We can solve this when we have confliction...

**Comment by XuetaoLiu-CLS** [comment]
Looks good other than a small change on the detector pane icon

Time Estimate: 5 minutes

**Pull request merged**